### PR TITLE
Use gulp to start the pattern lab server and watch for changes

### PR DESCRIPTION
**Pull request by daggerhart**
This addresses #22 

Changes:

* New `gulp-shell` module.
* New `gulp-shell` task for starting the pattern lab server
* Pattern lab server task added to the default gulp task
* Updated readme to reflect the simplified developer workflow
* Removed an old version of `list-element~1.json` which was causing a PHP warning during pattern lab compilation. 

One more thing: I'm not sure why exactly, but when I run this task the compiled css file changes the order of the `h3.list-title` and `div.list-body`. There doesn't appear to be a difference in the output for these elements, just their order has changed (see below). 

new-css-order

I don't think this should have been an effect of my changes, but since it is the result after my changes I've included it in this PR.

Thoughts/feedback welcome

### CODE REVIEW

File: design-library/source/_patterns/02-molecules/02-list-element/list-element~1.json [hunk 1]
```diff
@@ -1,8 +0,0 @@
-{
```
Comment (nerdstein) [inline design-library/source/_patterns/02-molecules/02-list-element/list-element~1.json]: I'm curious why this file is deleted. It probably should be there to show the second list element

**Review by nerdstein — CHANGES_REQUESTED** [review]
This looks great and I love the enhancement.

I'm not sure why the one file is being removed though. Can you take a look?

**Comment by daggerhart** [comment]
This maybe a misunderstanding of patternlab, but I deleted that file because I was getting PHP warnings during generation.  

(Like this, but not exactly. This is the warning I just got while testing #24)
screen shot 2017-06-13 at 11 32 21 am

I googled around for that error but didn't find anything, so tracked that down to `PatternLab\PatternData\Rule::getPatternName`. Found that patternlab was getting confused about the filename `list-element~1.json` and interpreting by looking for a item named "1". 

(var_dump FTW)
screen shot 2017-06-13 at 11 37 36 am

Deleting the `list-element~1.json` file resolved this warning.

And since I have dealt with IDEs in the past that will save temporary files with the `~1` naming convention, I assumed this file was an accident that shouldn't have been in the repo. 

I will look at it further this evening to see which part of my thought process was the main misunderstanding, and find a better way to resolve the PHP warnings.

**Comment by daggerhart** [comment]
Verified that this was my misunderstanding and restored the file (PR updated).

For my own reference, this is a "pseudo pattern" - http://patternlab.io/docs/pattern-pseudo-patterns.html 

Not sure why I'm getting the PHP warning, but everything appears to work fine. Maybe we need to rename the file in the future so that the pseudo-name is more than just "1"? Not too worried about it. Live and learn.

**Comment by nerdstein** [comment]
We can certainly rename the variations if "1" is throwing a warning. I also get the warning but I ignored it because things seemed to "work". 

I'm going to merge this and we can resolve the error in a separate issue.

**Pull request merged**